File: VB/WindowsApplication3/Custom Ribbon/MyRibbonViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.Drawing

Namespace DXSample
	Public Class MyRibbonViewInfo
		Inherits RibbonViewInfo
		Public Sub New(ByVal ribbonControl As RibbonControl)
			MyBase.New(ribbonControl)
		End Sub
		Protected Overrides Function CreateItemCalculator() As RibbonItemViewInfoCalculator
			Return New MyRibbonItemViewInfoCalculator(Me)
		End Function

		Public Shadows ReadOnly Property Manager() As MyRibbonBarManager
			Get
				Return TryCast(MyBase.Manager, MyRibbonBarManager)
			End Get
		End Property
		Protected Overrides Function CreatePanelInfo() As RibbonPanelViewInfo
			Return New MyRibbonPanelViewInfo(Me)
		End Function


	End Class
End Namespace

File: CS/WindowsApplication3/Custom Ribbon/MyRibbonViewInfo.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.Drawing;

namespace DXSample {
    public class MyRibbonViewInfo : RibbonViewInfo
    {
        public MyRibbonViewInfo(RibbonControl ribbonControl) : base(ribbonControl) { }
        protected override RibbonItemViewInfoCalculator CreateItemCalculator()
        {
            return new MyRibbonItemViewInfoCalculator(this);
        }

        public new MyRibbonBarManager Manager
        {
            get { return base.Manager as MyRibbonBarManager; }
        }
        protected override RibbonPanelViewInfo CreatePanelInfo()
        {
            return new MyRibbonPanelViewInfo(this);
        }
    }
}